exchanges/bitfinex/bitfinex.go

```diff
@@ -163,37 +158,32 @@ func symbolMarginInfo(data []interface{}) ([]MarginInfoV2, error) {
 	resp := make([]MarginInfoV2, len(data))
 	for x := range data {
 		var tempResp MarginInfoV2
-		tempData, ok := data[x].([]interface{})
+		marginInfo, ok := data[x].([]any)
 		if !ok {
-			return nil, fmt.Errorf("%w for all sym", errTypeAssert)
+			return nil, common.GetTypeAssertError("[]any", data[x], "MarginInfo")
 		}
 		var check bool
-		tempResp.Symbol, check = tempData[1].(string)
-		if !check {
-			return nil, fmt.Errorf("%w for symbol data", errTypeAssert)
+		if tempResp.Symbol, check = marginInfo[1].(string); !check {
+			return nil, common.GetTypeAssertError("string", marginInfo[1], "Symbol")
 		}
-		tempFloatData, check := tempData[2].([]interface{})
+		pairMarginInfo, check := marginInfo[2].([]any)
 		if !check {
-			return nil, fmt.Errorf("%w for symbol data", errTypeAssert)
+			return nil, common.GetTypeAssertError("[]any", marginInfo[2], "MarginInfo.Symbol")
 		}
-		if len(tempFloatData) < 4 {
+		if len(pairMarginInfo) < 4 {
 			return nil, errors.New("invalid data received")
 		}
-		tempResp.TradableBalance, ok = tempFloatData[0].(float64)
-		if !ok {
-			return nil, fmt.Errorf("%w for TradableBalance", errTypeAssert)
+		if tempResp.TradableBalance, ok = pairMarginInfo[0].(float64); !ok {
+			return nil, common.GetTypeAssertError("float64", pairMarginInfo[0], "MarginInfo.Symbol.TradableBalance")
 		}
-		tempResp.GrossBalance, ok = tempFloatData[1].(float64)
-		if !ok {
-			return nil, fmt.Errorf("%w for GrossBalance", errTypeAssert)
+		if tempResp.GrossBalance, ok = pairMarginInfo[1].(float64); !ok {
+			return nil, common.GetTypeAssertError("float64", pairMarginInfo[1], "MarginInfo.Symbol.GlossBalance")
 		}
-		tempResp.BestAskAmount, ok = tempFloatData[2].(float64)
-		if !ok {
-			return nil, fmt.Errorf("%w for BestAskAmount", errTypeAssert)
+		if tempResp.BestAskAmount, ok = pairMarginInfo[2].(float64); !ok {
+			return nil, common.GetTypeAssertError("float64", pairMarginInfo[2], "MarginInfo.Symbol.BestAskAmount")
 		}
-		tempResp.BestBidAmount, ok = tempFloatData[3].(float64)
-		if !ok {
-			return nil, fmt.Errorf("%w for BestBidAmount", errTypeAssert)
+		if tempResp.BestBidAmount, ok = pairMarginInfo[3].(float64); !ok {
+			return nil, common.GetTypeAssertError("float64", pairMarginInfo[3], "MarginInfo.Symbol.BestBidAmount")
```

Comment:
Shouldn't this be `MarginInfo.Data...` MarginInfo.Symbol is defined ln166 and also https://docs.bitfinex.com/reference/rest-auth-info-margin

Reply:
done